### ISSUE TITLE
[suggestion] zoom image when clicking on  image

### ISSUE THREAD

**Issue by avsingh999** [publiclab/plots2#5633]
I would like to suggest a new feature to add when clicking on the image then image zoom Like Medium.
thank you : )

**Comment by avsingh999** [comment]
@publiclab/reviewers @jywarren @SidharthBansal @gauravano @sashadev-sky  what do you think. please review me 
thanks : )

**Comment by sashadev-sky** [comment]
I didn't know how Medium does it so here is a link to a medium article for everyone for quick reference: https://elemental.medium.com/femtech-startups-are-finally-innovating-for-menopause-5d16bd1fdb3d. You can click on the image to zoom once (image proportionally takes up entire viewport) and a following click would toggle this (unzoom), or just on scroll it unzooms for you automatically. 

Compare to PL: https://publiclab.org/notes/Bronwen/04-29-2019/prototyping-a-tablet-microscope-like-the-cell-phone-microscope-but-bigger currently what we do is open a new tab on image click, where the image will take up the entire viewport and then you can also zoom once more. 

I think @avsingh999's idea looks a lot nicer! is this something you would want to implement yourself? 

Can we think of a situation where we would miss the additional zoom level provided by the PL implementation? In the PL example I provide I don't find it useful.

**Comment by gautamig54** [comment]
We already have a similar feature. On clicking on an image, it opens up separately. Then we can simply use control-plus or control-minus to zoom in and zoom out. So is this feature required? 

![ezgif com-video-to-gif (3)](https://user-images.githubusercontent.com/35326753/56956142-ed132c80-6b60-11e9-802a-18aea3e07ce2.gif)

**Comment by avsingh999** [comment]
Hi @gautamig54 @sashadev-sky I was thinking about these images in the notes https://publiclab.org/notes/Bronwen/04-29-2019/prototyping-a-tablet-microscope-like-the-cell-phone-microscope-but-bigger

example. 
![pub26](https://user-images.githubusercontent.com/25536022/56958743-c9ec7b00-6b68-11e9-921e-b4a5915a594d.png)

**Comment by avsingh999** [comment]
@sashadev-sky yeah, you are right It's not more useful but I was giving my Idea to better UI for PL.
thank you : )

Sorry for the late reply I am busy in my end sem exam.

**Comment by avsingh999** [comment]
@sashadev-sky @gautamig54  If you think it's not important and not useful you can close this issue.
thanks :smile:

**Comment by sashadev-sky** [comment]
Personally, I think it is an improvement on the UI. if you want to take this on, I say go for it. It looks a lot nicer than opening a new a tab with the image in my opinion. 

So, if you want to take this on, I will assign you to it! but if this is just an issue to float around, I don't think it is worth it to do such as we already have so many issues in plots2 to resolve. 

But I am totally with this update 🙋and I will oversee it as reviewer if you do it :)

**Comment by jywarren** [comment]
I was thinking about this, and I think we might think about the following case:

Many images now use a URL schema like: `![filename.jpg](/i/12345.jpg)` - this in Markdown. That's what our `images_controller` currently puts out. What if we auto-recognized that style, and, using only JavaScript, linked such images (if not already linked) to their full-res originals, which in this case would be at: `/i/12345.jpg?size=original` -- or for short, `/i/12345.jpg?s=o`?

This could be a relatively simple function that uses jQuery to scan for images that are /not/ in a link, and compare their `src` to a regex pattern, and when they match, wrap them in a link. Or we could look for a `click` event and respond by opening a new image with `window.open` (i think?)

**Comment by imajit** [comment]
@cesswairimu @jywarren  I think when one clicks on the image, we can have something similar to https://getbootstrap.com/docs/4.0/components/modal/ to pop up the image. Can I work on this?

**Comment by KarishmaVanwari** [comment]
Is this issue open? Is yes, I would like to work on this? @jywarren @sashadev-sky
